FilecoinFoundationWeb/filecoin-foundation
# Adding Github_Token

Automating repository scripts with GitHub Actions

## 📝 Description

Please include a summary of the changes. Provide context and motivation for the change, and describe what problem it solves.
Adding a new GitHub Actions workflow to automate project scripts and improve efficiency. This change provides a streamlined method for running Node.js scripts directly in the repository, utilizing GitHub's built-in GITHUB_TOKEN for secure authentication. This update enhances the project’s automation capabilities while maintaining best security practices.

- **Type:** Bug fix / New feature / Documentation / Refactor
New feature
## 🛠️ Key Changes

Added a new YAML workflow file for running Node.js scripts using GitHub Actions.

Configured GITHUB_TOKEN as a global environment variable for all scripts within the job.

Set up steps for checking out the repository, setting up Node.js, installing dependencies, and running custom scripts.



## 🧪 How to Test

Setup: Ensure that the GITHUB_TOKEN environment variable is correctly configured within GitHub Actions.

Steps to Test:

Push the workflow file to the main branch or trigger it manually.

Confirm that the workflow runs successfully and executes the desired scripts.

Check the workflow logs for any errors or warnings.

Expected Results: The workflow should run each script with no authentication errors, leveraging GITHUB_TOKEN securely.

Additional Notes: Ensure that any script requiring authentication references process.env.GITHUB_AUTH_TOKEN.

## 📸 Screenshots

Attach if there are UI changes.

## 🔖 Resources

Feel free to share any references to documentation, libraries, blog posts, or other resources that you consulted or used during the implementation of the changes.
